src/StudyEntities/Candidate/CandID.php

```diff
@@ -64,7 +64,7 @@ public function getType(): string
      */
     protected function validate(string $value): bool
     {
-        $pattern = sprintf("/[0-9]{%s}/", self::LENGTH);
+        $pattern = sprintf("/^[0-9]{%s}$/", self::LENGTH);
 
         return preg_match($pattern, $value) === 1 &&
             intval($value) >= self::MIN_VALUE;
```

Comment:
If I was writing this for the first time I'd probably do the following instead of using a regex:

```php
return mb_strlen($value) === self::LENGTH 
    && \Utility::valueIsPositiveInteger($value)
    && intval($value) >= self::MIN_VALUE;
```

I'm sure this fix is good though.

Reply:
But it 
    spread 
    over several
    lines.
Arguably, a regex is not easier to read.

Reply:
It's already a regex, the expression is just wrong.